plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogParser.java

```diff
@@ -267,10 +267,10 @@ private static Optional<LastCheckpoint> tryReadLastCheckpoint(TrinoFileSystem fi
         }
     }
 
-    public static long getMandatoryCurrentVersion(TrinoFileSystem fileSystem, String tableLocation)
+    public static long getMandatoryCurrentVersion(TrinoFileSystem fileSystem, String tableLocation, long readVersion)
             throws IOException
     {
-        long version = readLastCheckpoint(fileSystem, tableLocation).map(LastCheckpoint::getVersion).orElse(0L);
+        long version = readVersion;
```

Comment:
Nit: rename method parameter to get rid of this assignment

Reply:
I think this assignment is intentional because it's mutable object.